sbc_compassion/tools/import_letter_functions.py

```diff
@@ -159,15 +274,42 @@ def _find_qrcode(env, img, MULTIPAGE, file_,
             int(left), int(top), int(width), int(height)])
         # if the QR code is found
         if qrcode is not None:
+            right = int(left)+int(width)
+            bottom = int(top) + int(height)
+            test_img = cv2.imread(file_png)[int(top):bottom, int(left):right]
+            test_data = readEncode(test_img)
             # save it for zxing
             cv2.imwrite(file_png, img)
             # replace the image by the returned one
             data = _save_img(MULTIPAGE, file_, img)
         else:
-            cv2.imwrite(file_png, img[::-1, ::-1])
+            img = img[::-1, ::-1]
+            cv2.imwrite(file_png, img)
             qrcode = zx.decode(file_png, try_harder=True)
-    if qrcode is not None and 'XX' in qrcode.data:
-        partner_id, child_id = qrcode.data.split('XX')
+
+    partner_id, child_id = decodeBarcode(env, qrcode)
+
+    dict_odoo = {
+        'partner_id': partner_id,
+        'child_id': child_id,
+    }
+    if test:
+        dict_odoo.update({
+            'qr_preview': test_data
+        })
+    return dict_odoo, data, img
+
+
+def decodeBarcode(env, barcode):
+    """ Split the barcode and return the id of the partner and the child.
+    If the partner is not found, return None (same for the child).
+    :param env: Odoo environment
+    :param Barcode barcode:
```

Comment:
It is not a Tuple, but a class (see zxing.Barcode).
In the test case, I am just using a namedtuple in order to mimic the behavior of the barcode